provider/core/src/data_provider.rs

```diff
@@ -92,17 +95,52 @@ pub struct DataResponseMetadata {
 /// use icu_provider::prelude::*;
 /// use std::borrow::Cow;
 ///
-/// let payload = DataPayload { cow: Some(Cow::Borrowed("Demo")) };
+/// let payload = DataPayload { cow: Cow::Borrowed("Demo") };
 ///
-/// assert_eq!("Demo", payload.borrow().expect("Payload is present"));
+/// assert_eq!("Demo", &*payload);
 /// ```
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, PartialEq)]
 pub struct DataPayload<'d, T>
 where
     T: ToOwned + ?Sized,
     <T as ToOwned>::Owned: Debug,
 {
-    pub cow: Option<Cow<'d, T>>,
+    pub cow: Cow<'d, T>,
```

Comment:
`question`: is there a value in keeping it `pub`? It seems like an implementation detail that it is a `Cow` and the fact that the field is named after type follows that. Could we just use derefs/asrefs/borrows to get to the data?

Reply:
This is my next step.  I am making it non-pub in #730.